persistence/trees/trees.go

```diff
@@ -210,7 +213,9 @@ func (t *treeStore) getStateHash() string {
 	// create an order-matters list of roots
 	roots := make([][]byte, 0)
 	for tree := merkleTree(0); tree < numMerkleTrees; tree++ {
-		roots = append(roots, t.merkleTrees[tree].Root())
+		root := t.merkleTrees[tree].Root()
+		t.logger.Debug().Msgf("getStateHash root hash: %s", root)
```

Comment:
The comment isn't very clear / striaghtforward. Can you also add the tree we're logging it for and add a detail related to why we're doing it?

`t.logger.Debug().Msgf("getStateHash computing new root hash. Root for %s is %s", tree, root)`